service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/update/RetrieveControlledBigQueryDatasetWithoutLifetimeStep.java

```diff
@@ -0,0 +1,40 @@
+package bio.terra.workspace.service.resource.controlled.flight.update;
+
+import static bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys.ControlledResourceKeys.CONTROLLED_BIG_QUERY_DATASETS_WITHOUT_LIFETIME;
+
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+import bio.terra.workspace.db.ResourceDao;
+import bio.terra.workspace.service.resource.controlled.model.ControlledResource;
+import bio.terra.workspace.service.workspace.model.CloudPlatform;
+import java.util.List;
+
+public class RetrieveControlledBigQueryDatasetWithoutLifetimeStep implements Step {
+  private final CloudPlatform cloudPlatform;
+  private final ResourceDao resourceDao;
+
+  public RetrieveControlledBigQueryDatasetWithoutLifetimeStep(
+      CloudPlatform cloudPlatform, ResourceDao resourceDao) {
+    this.cloudPlatform = cloudPlatform;
+    this.resourceDao = resourceDao;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext context) throws InterruptedException, RetryException {
+    List<ControlledResource> controlledBigQueryDatasets =
```

Comment:
it's guaranteed to return List<ControlledGcpBqDataset> here right?

Reply:
Yes, now it does.